### PR TITLE
sink/mysql(ticdc): add comments

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5891

### What is changed and how it works?
Add a comment to mysql explaining how to detect conflicts

#### Tests 

 - No code

##### Will it cause performance regression or break compatibility?
No 
##### Do you need to update user documentation, design documentation or monitoring documentation?
No 
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/mysql/causality.go [hunk 1]
```diff
@@ -53,21 +60,34 @@ func (c *causality) reset() {
 // detectConflict detects whether there is a conflict
 func (c *causality) detectConflict(keys [][]byte) (bool, int) {
```
Comment (nongfushanquan) [inline cdc/sink/mysql/causality.go:61]: please add some comments about how to use the return value ( bool , int ).

**Review by asddongmen — APPROVED** [review]

File: cdc/sink/mysql/causality.go [hunk 2]
```diff
@@ -53,21 +60,34 @@ func (c *causality) reset() {
...
+			if conflictingWorkerIndex != workerIndex {
+				return true, multipleConflicting
 			}
 		}
 	}
 
-	return firstIdx != -1, firstIdx
+	// 1) no conflict
+	// 2) conflict with the same worker
+	return conflictingWorkerIndex != noConflicting, conflictingWorkerIndex
```
Comment (CharlesCheung96) [inline cdc/sink/mysql/causality.go:90]: Maybe better to separate conflictingWorkerIndex to conflictFlag and workerIndex?

**Comment by lance6716** [comment]
maybe link https://pingcap.com/zh/blog/tidb-binlog-source-code-reading-8

File: cdc/sink/mysql/causality.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sink/mysql/causality.go:67]: Done

File: cdc/sink/mysql/causality.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/sink/mysql/causality.go:96]: I think it's really just coupled together, and the fact that we distinguish it internally doesn't change the semantics of that function. So I don't think it's too much of a problem to just leave it that way. Unless we're aiming to refactor it. Then we should really redesign the API.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**